src/main/antlr/org/hibernate/sqm/parser/hql/internal/antlr/HqlParser.g4

```diff
@@ -87,10 +88,26 @@ orderingSpecification
 	;
 
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// LIMIT/OFFSET clause
+
+limitClause
+	: LIMIT parameterOrNumberLiteral
+	;
+
+offsetClause
+	: OFFSET parameterOrNumberLiteral
+	;
+
+parameterOrNumberLiteral
+	: parameter
+	| INTEGER_LITERAL
+	;
```

Comment:
In regards to the query-plan cache... If we support parameters here, IMO in cases where the database does not support that we should simply not cache the query plan.  I already have a bunch of checks for cases where we should not be caching the query plans such as multi-values parameters, etc.

However, I am not convinced we should support parameters here.  @dreab8, @Naros What are y'alls thoughts on this?

At the minimum, if the user specifies LIMIT/OFFSET using parameters and the database does not support that we should give a big fat warning.